src/main/java/com/WebController/EventController.java

```diff
@@ -25,18 +24,33 @@
     @Autowired
     private AssociationRepository assoicationRepo;
 
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ConfirmationTokenRepository confirmationTokenRepository;
+
+    @Autowired
+    private EmailSenderService emailSenderService;
+
     @GetMapping("/")
     public String homePage() {
         return "index";
     }
 
+    /**
+     * Handler for patients creation.
+     * User create a patient in front end and associate patient with multiple events.
+     * Data posted with JSON string and parsed to Event objects directly.
+     * Event duplication should be avoided.
+     */
     @PostMapping(value = "/patientinfo", consumes = "application/json")
     @ResponseBody
     public String receivePatient (@RequestBody List<Event> events){
-        Patient tempPatient = new Patient();
+        Patient tempPatient = new Patient();//created patient
         List<Event> eventList = new ArrayList<>();
-        Iterable<Event> allEvents = this.eventRepo.findAll();
-        allEvents.forEach(eventList::add);
+        Iterable<Event> allEvents = this.eventRepo.findAll();//pull all events from DB
+        allEvents.forEach(eventList::add);//add them to the list
         PatientEventAssociation association;
         for (Event e: events){
             int index = eventList.indexOf(e);
```

Comment:
why use indexOf() instead of contains()? Also, shouldn't there be a more efficient way of checking the presence of the event using the Repo instead of pulling the entire repo into a list? This is not going to scale well.

Reply:
The reason to use "indexOf()" here is to retrieve the existing event, modify and save it directly. 
I will use "findAllByNameAndDate()" to pull a list of events that have the same name and date, then use .equal() to check the address.